File: BackEnd/SpringBoot/product/src/main/java/com/example/productsservice/entity/product.java
package com.example.productsservice.entity;

import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Transient
    private Category category;
    private String name;
    private String description;
    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Image> images;
    private double price;
    private Long categoryID;
    private String sellerID;
    private Date created_at;
    private Date updated_at;
    private Date deleted_at;

}

File: FrontEnd/WebVersion/swap-shop/libs/products/src/lib/models/product.ts
import { Category } from "./category";
import { Image } from "./image";

export class product{
    id?: string;
    category?: Category;
    name?: string;
    description?: string;
    images?: Image[];
    price?:number;
    sellerID?: string;
    categoryID?: number;
    created_at?: Date;
    updated_at?: Date;
}
